web/html/src/manager/virtualization/guests/console/guests-console.js

```diff
@@ -53,6 +62,10 @@ class GuestsConsole extends React.Component<Props, State> {
     this.pageUnloading = false;
     this.openVirtSocket();
 
+    // Schedule token refresh a minute before its expiration
+    const tokenExp = getTokenExpiration(props.token);
+    this.intervalId = window.setInterval(this.refreshToken, tokenExp - Date.now() - 60000);
+
```

Comment:
I don't seem to grasp the refreshing logic completely:
- there is a periodic interval refreshing the token that should run once every X seconds (where X should be the lifetime of the token)
- but down there, we also have a timeout that is responsible for scheduling the "next" token refresh during a token refresh.

Shouldn't either of these be enough:
1. Only have the periodic interval, don't use the timeout below.
1. Use a timeout here (instead of an interval). It would schedule a one-time token refresh just before the expiration of the token. But in this one-time refresh, another refresh is scheduled by the code below.

Reply:
arf, right I forgot to remove the timeout